include/wx/private/uilocale.h

```diff
@@ -42,6 +42,9 @@ class wxUILocaleImpl
     // It may return NULL in case of failure.
     static wxUILocaleImpl* CreateForLanguage(const wxLanguageInfo& info);
 
+    // Platform dependent comparison of two strings, for a locale specified by wxLocaleIdent.
+    static int CmpLocalizedStrings(const wxString& lhs, const wxString& rhs, const wxLocaleIdent& locale_id);
```

Comment:
If we have it as static method, "Localized" in the name is probably redundant, as in `wxUILocale::CmpLocalizedString()`, so we could call it just `CompareStrings()` instead.

If we make it a global function, then `wxCmpLocalizedStrings()` is a good name, however.

Also, naming convention nitpick:
```suggestion
    // Platform dependent comparison of two strings, for a locale specified by wxLocaleIdent.
    static int CmpLocalizedStrings(const wxString& lhs, const wxString& rhs, const wxLocaleIdent& localeId);
```